digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/util/assetUpdater/LaneUpdater.scala

```diff
@@ -0,0 +1,414 @@
+      }
+    }
+
+    def saveSplitChanges(laneSplits: Seq[LaneSplit]): Unit = {
+      val lanesToCreate = laneSplits.flatMap(_.lanesToCreate).toSet
+      val lanesToExpire = laneSplits.map(_.originalLane).toSet
+
+      logger.info("Creating " + lanesToCreate.size + " new lanes due to split on link ids: " + lanesToCreate.map(_.linkId).mkString(", "))
+      val createdLaneIdsWithOldLanes = laneSplits.map(split => {
+        val createdLaneIds = split.lanesToCreate.map(laneToCreate => {laneService.createWithoutTransaction(laneToCreate, AutoGeneratedUsername.generatedInUpdate)})
+        (createdLaneIds, split.originalLane)
+      })
+
+      // Each old lane must have history row referencing to each new lane created from it
+      logger.info("Expiring " + lanesToExpire.size + " old lanes due to split on link ids: " + lanesToExpire.map(_.linkId).mkString(", "))
+      createdLaneIdsWithOldLanes.foreach(createdLaneIdAndOldLane => {
+        val createdLaneIds = createdLaneIdAndOldLane._1
+        val originalLane = createdLaneIdAndOldLane._2
+        createdLaneIds.foreach(newId => {
+          laneService.moveToHistory(originalLane.id, Some(newId), expireHistoryLane = true, deleteFromLanes = false, AutoGeneratedUsername.generatedInUpdate)
+        })
+      })
+
+      // After history is created, delete lanes
+      logger.info("Deleting " + lanesToExpire.size + " old lanes due to split on linkIds: " + lanesToExpire.map(_.linkId).mkString(", "))
+      lanesToExpire.foreach(laneToExpire => {
+        laneService.dao.deleteEntryLane(laneToExpire.id)
+      })
+    }
+
+    // Expire lanes marked to be expired
+    val laneIdsToExpire = changeSet.expiredLaneIds.toSeq
+    if (laneIdsToExpire.nonEmpty) {
+      logger.info("Expiring ids: " + laneIdsToExpire.mkString(", "))
+      laneIdsToExpire.foreach(laneId => {
+        laneService.moveToHistory(laneId, None, expireHistoryLane = true, deleteFromLanes = true, AutoGeneratedUsername.generatedInUpdate)
+      })
+    }
+
+    // Create generated new lanes
+    if (changeSet.generatedPersistedLanes.nonEmpty){
+      logger.info(s"${changeSet.generatedPersistedLanes.size} lanes created for ${changeSet.generatedPersistedLanes.map(_.linkId).toSet.size} links")
+      laneService.createMultipleLanes(changeSet.generatedPersistedLanes, AutoGeneratedUsername.generatedInUpdate)
+    }
+
+    // Create new lanes, and expire old lanes due to road link splits
+    if (changeSet.splitLanes.nonEmpty) {
+      saveSplitChanges(changeSet.splitLanes)
+    }
+
+    // Adjust m-values and side codes for lanes marked to be adjusted
+    if (changeSet.positionAdjustments.nonEmpty) {
+      logger.info("Saving SideCode/M-Value adjustments for lane Ids:" + changeSet.positionAdjustments.map(a => "" + a.laneId).mkString(", "))
+      saveLanePositionAdjustments(changeSet.positionAdjustments)
+    }
+  }
+
+
+
+  def updateLanes(): Unit = {
+    val allRoadLinkChanges = roadLinkChangeClient.getRoadLinkChanges()
```

Comment:
eikös tässä pitäisi hakea edellisen päivän muutokset, Alla oleva koodi kyl hakee kaikki muutokset mutta getRoadLinkChanges oli parametrina päivä
Missä merkataan samuutus onnistuneeksi ja tallennetaan, se päivämä määrä jotta voidaan taas viimeismmästä onnistuunesta samuutuksesta eteenpäin?

Reply:
Lisätty samuutusSuccess päivitys samuutuksen loppuun